### PR TITLE
Design and engineering collaboration

### PR DESCRIPTION
**Pull request by rrhyne**
Post principles on design and engineering collaborate.

### CODE REVIEW

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Both Design and Engineering should have a clear understanding of the goals, target audience, business requirements, technical constraints, and desired outcomes.
+
+## Early involvement & regular communication
+
+Design and Engineering should both be involved in a project as early as possible so that each specialty can provide insight as questions come up about feasibility, performance considerations, and UX challenges. Because this is the overlap of two domains, tight feedback loops should be a high priority. Often design introduces elements that would require additional engineering or issues are discovered in the build cycle that impact design decisions. Regular communication helps resolve these issues.
+
+## Standard set of tools and tooling
+
+We use tools that allow for clear communication of ideas between parties, and that both can use. For example, design users the same issue tracker as engineering. Additionally, design and engineering collaborate to keep tooling like design systems in sync, so that designs use components that are the same as their code base counterparts.
```
Comment (rrhyne) [inline content/departments/product/design/design-and-engineering-collaboration.md]:
```suggestion
We use tools that allow for clear communication of ideas between parties, and that both can use. For example, designers use the same issue tracker as engineering. Additionally, design and engineering collaborate to keep design systems in sync so that designs use components that are the same as their codebase counterparts.
```

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+We use tools that allow for clear communication of ideas between parties, and that both can use. For example, designers use the same issue tracker as engineering. Additionally, design and engineering collaborate to keep design systems in sync so that designs use components that are the same as their codebase counterparts.
+
+We value consistency over novelty and diligently work to ensure new work utilizes our shared tooling. We ensure that updates to this tooling are carefully considered by both design and engineering.
+
+## Communicate and understand constraints
+
+Design should have a basic understanding of the technical constraints of the platform. Engineering should understand the importance of design decisions and their impact on the user experience.
+
+Both parties are encouraged to contribute to all sides of the discussion. Design should bring up technical considerations. Engineering should contribute UX ideas. Both should defer to the other in their area of expertise: Design should defer to Engineering on technical considerations, and Engineering should defer to Design on UX ideas.
```
Comment (rrhyne) [inline content/departments/product/design/design-and-engineering-collaboration.md:35]:
```suggestion
Both parties are encouraged to contribute to all sides of the discussion. Design should bring up technical considerations. Engineering should contribute UX ideas. Both should defer to the other in their area of expertise: Design should defer to Engineering on technical considerations, and Engineering should defer to Design on UX ideas.

## Design is what ships, not what is rendered in design software

To ship the best software, design and engineering collaborate from the beginning of a project to it's release. This collaboration starts with ideation to kick projects off on the strongest footing and finishes with comprehensive QA to provide multiple perspectives on the quality of the shipped product.
```

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Design is what ships, not what is rendered in design software
+
+To ship the best software, design and engineering collaborate from the beginning of a project to it's release. This collaboration starts with ideation to kick projects off on the strongest footing and finishes with comprehensive QA to provide multiple perspectives on the quality of the shipped product.
```
Comment (rrhyne) [inline content/departments/product/design/design-and-engineering-collaboration.md]:
```suggestion
To ship the best software, design and engineering collaborate from the beginning of a project to its release. This collaboration starts with ideation to kick projects off on the strongest footing and finishes with comprehensive QA to provide multiple perspectives on the quality of the shipped product.
```

**Review by taiyab — COMMENTED** [review]
Looks good, thanks for putting this together!

**Review by unknwon — APPROVED** [review]
🌟

**Review by camdencheek — APPROVED** [review]
I love this. Thanks for putting it together!

File: content/departments/product/design/index.md [hunk 4]
```diff
@@ -65,6 +65,7 @@ Product design at Sourcegraph is low-process and highly adaptive. Instead of des
 - [Evaluative research](./research/conducting-evaluative-research.md)
 - [Design and interaction guidelines](design-and-interaction-guidelines.md)
 - [Design collaboration and review](./design_collaboration_review/index.md)
+- [Collaboration with engineering](./design-and-engineering-collaboration.md)
```
Comment (ErikaRS) [inline content/departments/product/design/index.md:68]: Would it make sense to also link this in the "Processes" section of https://handbook.sourcegraph.com/departments/engineering/ so we have links from both sides?

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## We make time for design before engineering begins
+
+By beginning with design we help the entire team explore product requirements in the least expensive and most collaborative medium. When we all make time to collaborate in the design stage, we provide everyone with the ability to review the solution before development. This process should lead to stronger outcomes with less technical debt, as engineering’s systems design has a better chance of being correct on the first try.
+
+## We design together, throughout the project
+
+From planning to QA, everyone shapes the outcome of the product, each step involves some measure of design, product, and engineering input. When we have feedback from a wide range of disciplines, we produce a much better product.
+
+As a project proceeds, design and engineering collaborate to modify the effort as required by various constraints. Both parties are flexible and seek to understand tradeoffs to arrive at the best solution given available resources. We understand that at times both design and tech debt is generated during this process. We track this debt and seek to reduce it when time allows.
```
Comment (ErikaRS) [inline content/departments/product/design/design-and-engineering-collaboration.md]: Tiny nitpick: most of the time in this doc, Design and Engineering are capitalized when used as descriptions of the disciplines, but not here.

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+## We embrace incremental development
+
+To avoid huge projects that complicate matters for engineering, we strive to break projects into iterations that help us more efficiently build and deliver projects and test them with end users.
```
Comment (ErikaRS) [inline content/departments/product/design/design-and-engineering-collaboration.md]: Another Engineering that could be capitalized.

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+Both Design and Engineering should have a clear understanding of the goals, target audience, business requirements, technical constraints, and desired outcomes.
+
+## Early involvement & regular communication
+
+Design and Engineering should both be involved in a project as early as possible so that each specialty can provide insight as questions come up about feasibility, performance considerations, and UX challenges. Because this is the overlap of two domains, tight feedback loops should be a high priority. Often design introduces elements that would require additional engineering or issues are discovered in the build cycle that impact design decisions. Regular communication helps resolve these issues.
+
+## Standard set of tools and tooling
+
+We use tools that allow for clear communication of ideas between parties, and that both can use. For example, designers use the same issue tracker as engineering. Additionally, design and engineering collaborate to keep design systems in sync so that designs use components that are the same as their codebase counterparts.
```
Comment (ErikaRS) [inline content/departments/product/design/design-and-engineering-collaboration.md]: Or maybe it's about 50/50 whether or not they are capitalized 😉. Either way, "design and engineering collaborate" is another that should use whichever of lowercase or capital you decide on!

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+## Communicate and understand constraints
+
+Design should have a basic understanding of the technical constraints of the platform. Engineering should understand the importance of design decisions and their impact on the user experience.
+
+Both parties are encouraged to contribute to all sides of the discussion. Design should bring up technical considerations. Engineering should contribute UX ideas. Both should defer to the other in their area of expertise: Design should defer to Engineering on technical considerations, and Engineering should defer to Design on UX ideas.
+
+## Design is what ships, not what is rendered in design software
+
+To ship the best software, design and engineering collaborate from the beginning of a project to its release. This collaboration starts with ideation to kick projects off on the strongest footing and finishes with comprehensive QA to provide multiple perspectives on the quality of the shipped product.
```
Comment (ErikaRS) [inline content/departments/product/design/design-and-engineering-collaboration.md]: One more capitalization consistency pointer.

Comment (ErikaRS) [inline content/departments/product/design/design-and-engineering-collaboration.md]: The first 1.5 sentences of this cover the same area as the first two points. I think the value add of this point is the bit about making sure that the shipped product meets the quality bar. I'd love if you could rephrase this to make that point more prominent.

**Review by ErikaRS — COMMENTED** [review]
Thanks for writing this! Most of my comments are just copy-edit nitpicks about consistent capitalization, but I did have one comment on the substance of the last point.

**Review by eseliger — APPROVED** [review]

**Review by toolmantim — APPROVED** [review]

File: content/departments/product/design/index.md (hunk 4 above)
Comment (rrhyne) [reply content/departments/product/design/index.md:68]: Yes, I meant to ask about where on that side, thank you!

Comment (rrhyne) [reply content/departments/product/design/index.md:68]: Addressed [here](https://github.com/sourcegraph/handbook/pull/8215/files#diff-b2a725171075afa41533266a42e359a43fe13e858a7fe87e630aff12f6ec9c87R42). Thanks again!

File: content/departments/product/design/design-and-engineering-collaboration.md (hunk 5 above)
Comment (rrhyne) [reply content/departments/product/design/design-and-engineering-collaboration.md]: Ah, jeez, that was a bit of a shitshow. I went through and capitalized the departments while NOT capitalizing references to the practices. I hope I got it all right. 

Thanks for your diligence.

File: content/departments/product/design/design-and-engineering-collaboration.md (hunk 8 above)
Comment (rrhyne) [inline content/departments/product/design/design-and-engineering-collaboration.md]:
```suggestion
We must carefully collaborate in all phases of an effort, from design through to engineering and QA. This allows designers to help resolve contingencies in the build stages, and it provides engineers with much-needed backup in the QA process. With this kind of strong collaboration throughout, we'll improve the quality of both our design and engineering.
```

Comment (rrhyne) [reply content/departments/product/design/design-and-engineering-collaboration.md]: @ErikaRS wdyt?

File: content/departments/product/design/design-and-engineering-collaboration.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+## Communicate and understand constraints
+
+Design should have a basic understanding of the technical constraints of the platform. Engineering should understand the importance of design decisions and their impact on the user experience.
+
+Both parties are encouraged to contribute to all sides of the discussion. Design should bring up technical considerations. Engineering should contribute UX ideas. Both should defer to the other in their area of expertise: Design should defer to Engineering on technical considerations, and Engineering should defer to Design on UX ideas.
+
+## Design is what ships, not what is rendered in design software
+
+To ship the best software, Design and Engineering collaborate from the beginning of a project to its release. This collaboration starts with ideation to kick projects off on the strongest footing and finishes with comprehensive QA to provide multiple perspectives on the quality of the shipped product.
```
Comment (rrhyne) [inline content/departments/product/design/design-and-engineering-collaboration.md]:
```suggestion
The gap between what is envisioned and what the final release can be is often large. By maintaining strong collaboration and ownership from ideation through to the shared responsibility of QA, each discipline contributes to shipping design that delights our users. 
```

Comment (rrhyne) [reply content/departments/product/design/design-and-engineering-collaboration.md]: @ErikaRS wdyt? Note: GitHub seems to be having some issues, this is version 2 of this change, but I like it better!

**Review by danielmarquespt — APPROVED** [review]
